deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+from typing import Dict, List
+
+import numpy as np
+from overrides import overrides
+
+from .instance import TextInstance, IndexedInstance
+from ..data_indexer import DataIndexer
+
+
+class TextTuple:
+    def __init__(self, subject: str, verb: str, objects: List[str],
+                 context: str=None, time: str=None, loc: str=None, source: str=None):
+        self.subject = subject
+        self.verb = verb
+        self.objects = objects
+        self.context = context
+        self.time = time
+        self.location = loc
+        self.source = source
+
+    def __str__(self):
+        # NOTE: not currently displaying locaton, time, and source info
+        return "TextTuple( [\n\tsubj: {0} \n\tverb: {1} \n\tobjects: {2} \n\tcontext: {3}\n] )"
+
+    def to_text_list(self, object_handling: str="collapse", include_context: bool=True):
+        # Subject and verb
+        text_list = [self.subject, self.verb]
+        # Object(s)
+        if object_handling == "collapse":
+            text_list.append(" ".join(self.objects))
+        elif object_handling == "first":
+            if len(self.objects) > 0:
+                text_list.append(self.objects[0])
+        else:
+            raise Exception("Requested object_handling ({0}) not currently supported.".format(object_handling))
+        # Context
+        if include_context and self.context:
+            text_list.append(self.context)
+        # Currently, source, time and location aren't being used, but this can be changed in the future.
+
+        return text_list
+
+
+    @classmethod
+    def from_string(cls, input_string: str):
+        tuple_fields = input_string.lower().split("|")
+        num_fields = len(tuple_fields)
+        assert num_fields >= 2, "num_fields not >= 2" + input_string
+        subject = tuple_fields[0]
+        verb = tuple_fields[1]
+        objects = []
+        context, time, location, source = None, None, None, None
+        for element in tuple_fields[2:]:
+            if element.startswith("context:"):
+                context = element.split("context:")[1]
+            elif element.startswith("L:"):
+                location = element.split("L:")[1]
+            elif element.startswith("T:"):
+                time = element.split("T:")[1]
+            elif element.startswith("source:"):
+                source = element.split("source:")[1]
+            else:
+                objects.append(element)
+        return TextTuple(subject, verb, objects, context, time, location, source)
+
+
+class MCTuplesWithBackgroundInstance(TextInstance):
+    """
+    A MCTuplesWithBackgroundInstance is a kind of TextInstance that has text in multiple slots (stored
+    as a TextTuple), for both the answer choices and background information.
+    This can be used to store SVO triples.
+
+    Parameters
+    ----------
+    - answer_tuples: List[List[TextTuple]]
+        This is a list of length = num answer options.  Each answer has a list of TextTuples (made
```

Comment:
no need for the `=`. I feel like you're treating `num answer options` as a variable here --- if so, add double backticks and replace spaces with underscores. Else this might be better worded as `"The length of `answer_tuples` must be the same as the number of answer options"`.